src/lib/local-storage/container.js

```diff
@@ -0,0 +1,39 @@
+/* @flow */
+import constants from '../constants';
+import type { ContainerSummary } from '../../types';
+
+const { storage, oneHour } = constants;
+
+const getContainer = () : Object | null => {
+  const storedValue = window.localStorage.getItem(storage.paypalCrContainer);
+
+  if (storedValue) {
+    return JSON.parse(storedValue);
+  }
+
+  return null;
+};
+
+/* Returns an existing, non-expired container. Removes a container
+from localstorage if it has expired. */
+export const getValidContainer = () : ContainerSummary | null => {
+  const storedValue = getContainer();
+  const now = Date.now();
+
+  if (!storedValue || ((now - storedValue.createdAt) > oneHour)) {
```

Comment:
I'm wondering if we can do even lower than an hour. I think the considerations we have to balance are:
1. Allowing merchants to send as many offers as possible
2. Latency of getting the container

What would be the effect if we dropped this down to say, 15 minutes?

Reply:
This is a complete guess, but I'd wager you'd see a noticeable increase in the number of requests made to tagmanager.